app/services/proxy_status/puppetca.rb

```diff
@@ -0,0 +1,71 @@
+module ProxyStatus
+  class PuppetCA < Base
+    def certs
+      fetch_proxy_data('/certs') do
+        api.all.map do |name, properties|
+          SmartProxies::PuppetCACertificate.new([name.strip, properties['state'], properties['fingerprint'], properties["not_before"], properties["not_after"], self])
+        end.compact.sort_by!(&:valid_from)
+      end
+    end
+
+    def find(name)
+      certs.find{|c| c.name == name}
+    end
+
+    def find_by_state(state)
+      certs.select{|c| c.state == state}
+    end
+
+    def expiry
+      ca_cert = certs.first
+      if ca_cert.present?
+        ca_cert.expires_at
+      else
+        _("Could not locate CA certificate.")
+      end
+    end
+
+    def sign(cert)
+      Rails.cache.delete(cache_key+'/certs')
+      api.sign_certificate(cert)
+    end
+
+    def destroy(cert)
+      Rails.cache.delete(cache_key+'/certs')
+      api.del_certificate(cert)
+    end
+
+    def autosign
+      fetch_proxy_data('/autosign') do
+        api.autosign
+      end
+    end
+
+    def set_autosign(id)
+      Rails.cache.delete(cache_key+'/autosign')
+      api.set_autosign(id)
+    end
+
+    def del_autosign(id)
+      Rails.cache.delete(cache_key+'/autosign')
+      api.del_autosign(id)
+    end
+
+    def revoke_cache!
+      Rails.cache.delete(cache_key+'/certs')
+      Rails.cache.delete(cache_key+'/autosign')
+      super
```

Comment:
Right, it doesn't actually do anything as nothing is cached under just cache_key. removing.